src/component/Autocomplete.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+
+export interface AutocompleteProps {
+  suggestions: string[];
+}
+
+const Autocomplete: React.FC<AutocompleteProps> = ({ suggestions }) => {
+  const [inputValue, setInputValue] = useState('');
+  const [filteredSuggestions, setFilteredSuggestions] = useState<string[]>([]);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const userInput = e.currentTarget.value;

```

Comment:
check for null values  at second level. 

you may use like e.currentTarget?.value